lib/filemaker/model/builder.rb

```diff
@@ -42,10 +30,15 @@ def single(resultset, klass)
 
         record.keys.each do |fm_field_name|
           # record.keys are all lowercase
-          field = klass.find_field_by_name(fm_field_name)
+          field = object.class.find_field_by_name(fm_field_name)
           next unless field
 
-          object.public_send("#{field.name}=", record[fm_field_name])
+          if hydrate_directly
+            # Because we are using ActiveModel::Dirty, so we hydrate directly.
+            object.attributes[field.name] = field.coerce(record[fm_field_name])
```

Comment:
From a global search, I don't think I am using `Filemaker::Model::Builder.single` at all. I just keep using the collection and when I want to find the first object, the `Criteria#first` will still make use of the `Filemaker::Model::Builder.collection` instead. So maybe we can remove this module function.

Let me get back after my day's work to examine further.